src/plugins/data/public/query/query_string/dataset_service/types.ts

```diff
@@ -16,6 +22,20 @@ export interface DataStructureFetchOptions {
   paginationToken?: string;
 }
 
+export interface DatasetIndexedView {
+  name: string;
+}
+
+export interface DatasetIndexedViewsService {
+  getIndexedViews: (dataset: Dataset) => Promise<DatasetIndexedView[]>;
+  /**
+   * Returns the data source saved object connected with the data connection object
+   * @param dataConnectionId Id of the data-connection saved object for which we want the data-source
+   * @returns Data source saved object for the Collection/Cluster attached with the data-connection
+   */
+  getConnectedDataSource: (dataConnectionId: string) => Promise<SavedObject>;
```

Comment:
isn't this already present in the dataset.dataSource? This might be putting too much on the dataset service or another thing to maintain. if the devs need the access the data source then can just use the saved objects client directly.

Reply:
The data source contains info about the data connection itself, this is need to be able to use the data connection's attributes and fetch the matching data source (collection/cluster). It is added to the indexed view service and only the type lives here in the dataset_service

Reply:
could you check out this PR here? 

https://github.com/opensearch-project/OpenSearch-Dashboards/pull/8845

this might reduce the complexity of adding another service